### PR TITLE
Access control improvements.

### PR DESCRIPTION
**Pull request by sunsetsystems**
This continues work to fill in access control omissions.

Added these ACOs in the "patients" section:
  reminder    Patient Reminders (write,addonly optional)
  alert       Clinical Reminders/Alerts (write,addonly optional)
  disclosure  Disclosures (write,addonly optional)
  rx          Prescriptions (write,addonly optional)
  amendment   Amendments (write,addonly optional)
  lab         Lab Results (write,addonly optional)

Also added code, mostly in the demographics summary, to enforce these and some others.
Also modified acl.inc to always give the superuser access to everything.
Also updated menu code to disable items that the user does not have permission for.

### CODE REVIEW

File: acl_upgrade.php [hunk 1]
```diff
@@ -413,14 +421,78 @@
...
+  //Insert the 'rx' object from the 'patients' section into the Physicians group write ACL (added in 5.0.1)
+  updateAcl($doc_write, 'Physicians', 'patients', 'Patients', 'rx', 'Prescriptions (write,addonly optional)', 'write');
+  //Insert the 'amendment' object from the 'patients' section into the Physicians group write ACL (added in 5.0.1)
+  updateAcl($doc_write, 'Physicians', 'patients', 'Patients', 'amendment', 'Amendments (write,addonly optional)', 'write');
+  //Insert the 'lab' object from the 'patients' section into the Physicians group write ACL (added in 5.0.1)
+  updateAcl($doc_write, 'Physicians', 'patients', 'Patients', 'lab', 'Lab Results (write,addonly optional)', 'write');
+
+  //DONE with upgrading to this version
+  $acl_version = $upgrade_acl;
+}
```
Comment (bradymiller) [inline acl_upgrade.php]: woohooo, after some years, this mechanism finally gets used :)

File: interface/main/tabs/js/menu_analysis.js [hunk 2]
```diff
@@ -218,6 +218,8 @@ function menu_entry_from_jq(elem)
 var menu_entries=[];
 function analyze_menu()
 {
+    alert('I think you will never see this. --Rod'); // debugging
```
Comment (bradymiller) [inline interface/main/tabs/js/menu_analysis.js]: but I am seeing it here :)

Comment (bradymiller) [inline interface/main/tabs/js/menu_analysis.js]: btw, above is a bad joke. I have not tested this code.

File: library/acl.inc [hunk 3]
```diff
@@ -148,6 +154,11 @@
   function acl_check($section, $value, $user = '', $return_value = '') {
     if (! $user) $user = $_SESSION['authUser'];
 
+    // Superuser always gets access to everything.
+    if (($section != 'admin' || $value != 'super') && acl_check('admin', 'super', $user)) {
+      return TRUE;
+    }
```
Comment (bradymiller) [inline library/acl.inc]: very good to do this

**Comment by bradymiller** [comment]
Very nice to see this. Code looks good to me. I didn't test it, but am guessing it's been tested extensively already.

**Comment by sunsetsystems** [comment]
Well that was easy. Thanks Brady!

**Comment by sharonco-matrix** [comment]
That was really fast!!!! Great work ! and thanks

**Pull request merged**